frappe/email/doctype/email_alert/email_alert.py

```diff
@@ -171,13 +142,55 @@ def get_attachment(doc):
 
 		attachments = get_attachment(doc)
 
-		frappe.sendmail(recipients=recipients, subject=subject,
-			message= frappe.render_template(self.message, context),
-			reference_doctype = doc.doctype,
-			reference_name = doc.name,
-			attachments = attachments,
-			print_letterhead = ((attachments
-				and attachments[0].get('print_letterhead')) or False))
+		if self.channel == 'Email':
+			recipients = []
+			for recipient in self.recipients:
+				if recipient.condition:
+					if not frappe.safe_eval(recipient.condition, None, context):
+						continue
+				if recipient.email_by_document_field:
+					if validate_email_add(doc.get(recipient.email_by_document_field)):
+						recipient.email_by_document_field = doc.get(recipient.email_by_document_field).replace(",", "\n")
+						recipients = recipients + recipient.email_by_document_field.split("\n")
+
+					# else:
+					# 	print "invalid email"
+				if recipient.cc and "{" in recipient.cc:
+					recipient.cc = frappe.render_template(recipient.cc, context)
+
+				if recipient.cc:
+					recipient.cc = recipient.cc.replace(",", "\n")
+					recipients = recipients + recipient.cc.split("\n")
+
+				#For sending emails to specified role
+				if recipient.email_by_role:
+					emails = get_emails_from_role(recipient.email_by_role)
+
+					for email in emails:
+						recipients = recipients + email.split("\n")
+
+			if not recipients:
+				return
+
+			recipients = list(set(recipients))
+
+			frappe.sendmail(recipients=recipients, subject=subject,
+				message= frappe.render_template(self.message, context),
+				reference_doctype = doc.doctype,
+				reference_name = doc.name,
+				attachments = attachments,
+				print_letterhead = ((attachments
+					and attachments[0].get('print_letterhead')) or False))
+
+		elif self.channel == 'Slack':
+			slack_webhook_url = self.slack_webhook_url
```

Comment:
should this be part of the Slack Setup?

Reply:
Sure I'll refactor it

Reply:
Actually the user needs to choose it's channel manually: one notification can be sent to one channel, the other one to another channel.

I could of course drop this variable which is not useful and write it like:

```
send_slack_message(
				webhook_url=self.slack_webhook_url,
				message=frappe.render_template(self.message, context),
				reference_doctype = doc.doctype,
				reference_name = doc.name
			)
```